lab1/README.md

```diff
@@ -152,4 +152,284 @@ func main() {
+* **Read-Only Channel**:
+    * Also known as **Receive-Only Channels**
+    * Only allow read operations from the channel, no write operations
+    * Restrict goroutines to only read data from the channel, which can improve the program's readability
+    * Creation method:
+    ```go
+    c := make(<- chan int, 10) //notice the <- operator direction
+    c <- 1 //cannot write to c
+    fmt.Println(<-c)
+    ```
+* **Write-Only Channel**:
+    * Also known as **Send-Only Channel**
+    * Only allow write operations from the channel, no read operations
+    * Sending data to other goroutines and enhancing security
+    * Creation method:
+    ```go
+    c := make(chan <- int, 10)
+    c <- 1
+    ```
+## Select
+In the previous sections, we learned how to create channels and categorize them. We understand that when a single channel performs a send or receive operation, it is a blocking operation. When a program involves multiple channels for communication, Golang provides the select statement for this purpose. It allows a goroutine to wait on multiple communication operations simultaneously. Its usage is similar to **`switch`**, relying on **`case`** and **`default`**.
+
+**`select`** is in blocking operations; it starts execution only when a `case` is ready. Unlike **`switch`**, **`select`** doesn't execute cases **in order** but **randomly chooses** one from the ready cases (channels). There are several characteristics：
+* **`select`** can only work with channels; using other types will result to error
+* If a channel has no value to read, it causes a panic
+* When none of the cases are ready, **`select`** executes the **`default`**. Notice that without a **`default`** case, select will be blocked if none of the cases are ready
+```go
+func main() {
+    ch := make(chan int, 1)
+
+    select {
+    case <-ch:
+        fmt.Println("random 01")
+    case <-ch:
+        fmt.Println("random 02")
+    default:
+        fmt.Println("exit")
+    }
+}
+```
+In the example above, due to **`ch`** does not have a value, none of the cases are ready, so the **`default`** case will be executed directly, then printing "exit".
+### :tada: Setup Timeout mechanism
+Sometimes, we encounter situations where a goroutine takes too long to execute or gets into blocking. In such cases, we don't want the entire program to be blocked within **`select`**. To handle this, we can setup **`timeout`** with **`select`**:
+```go
+func main() {
+    timeout := make(chan bool, 1)
+    go func() {
+        time.Sleep(2 * time.Second)
+        timeout <- true
+    }()
+    ch := make(chan int)
+    select {
+    case <-ch:
+    case <-timeout:
+        fmt.Println("Open5GS")
+    case <-time.After(time.Second * 1): //Create a Read-Only Channel which will receive time.Time value
```

Comment:
Hi @KunLee76 
Should be careful when use timer, this sample program has memory leakage.
Reference: https://arangodb.com/2020/09/a-story-of-a-memory-leak-in-go-how-to-properly-use-time-after/

Please also add the article here for reference.

@andy89923 FYI